### PR TITLE
WIP: Update to VPR (including router refactor)

### PR DESCRIPTION
**Pull request by litghost**
This is VPR master + https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1126

FYI, the wip branches for this PR were made on https://github.com/litghost/vtr-verilog-to-routing/ not https://github.com/Symbiflow/vtr-verilog-to-routing/

### CODE REVIEW

**Comment by litghost** [comment]
@acomodi FYI

**Comment by litghost** [comment]
@HackerFoo I made the following change to the lookahead here: https://github.com/litghost/vtr-verilog-to-routing/commit/22077294cceb26369bc455dd7e1f4c81f735f021

I believe this is required because on upstream you changed the following assert from:
https://github.com/SymbiFlow/vtr-verilog-to-routing/blob/bb0ba83ea04d2c228460260247144151ac016f25/libs/libvtrutil/src/vtr_geometry.tpp#L181
to:
https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/edbc181ed606d376f9dbc0ecc724e17ac472c96d/libs/libvtrutil/src/vtr_geometry.tpp#L181

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/verilog-to-routing/vtr-verilog-to-routing">verilog-to-routing/vtr-verilog-to-routing</a></strong></div><div>Verilog to Routing -- Open Source CAD Flow for FPGA Research - verilog-to-routing/vtr-verilog-to-routing</div></blockquote>

**Review by acomodi — APPROVED** [review]
LGTM.
Two questions:
1. The wip branches can be pushed here, so that they we can use them for next master+wip generation, without having to solve conflicts?
2. https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1126 should this one get on a wip branch as well, or we just wait for it to get merged upstream?

Travis CI failed for a wrong golden result (the one obtained seems actually to be better then the golden one):
```
 k6_N10_mem32K_40nm.xml/stereovision3.v/common critical_path_delay: golden = 3.18721 result = 2.48622
[Fail] 
 k6_N10_mem32K_40nm.xml/stereovision3.v/common setup_WNS: golden = -3.18721 result = -2.48622
```

**Comment by litghost** [comment]
> 1. The wip branches can be pushed here, so that they we can use them for next master+wip generation, without having to solve conflicts?

After verilog-to-routing#1126 is merged, ya I'll push new wip branches to the Symbiflow fork.  I wanted to get something up, so you can test additional changes to VTR as needed.

> 2\. [verilog-to-routing#1126](https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1126) should this one get on a wip branch as well, or we just wait for it to get merged upstream?

Because of the weight of the change, I want to wait until it is merged upstream.

**Comment by litghost** [comment]
> Travis CI failed for a wrong golden result (the one obtained seems actually to be better then the golden one):

I'll fix that in a follow to this PR once changes land upstream.

**Review by HackerFoo — APPROVED** [review]

**Pull request closed without merging**
